src/main/kotlin/org/jitsi/nlj/rtp/RedAudioRtpPacket.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright @ 2018 - Present, 8x8 Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.nlj.rtp
+
+import org.jitsi.rtp.rtp.RedPacketBuilder
+import org.jitsi.rtp.rtp.RedPacketParser
+import java.lang.IllegalStateException
+
+class RedAudioRtpPacket(
+    buffer: ByteArray,
+    offset: Int,
+    length: Int
+) : AudioRtpPacket(buffer, offset, length) {
+
+    private var removed = false
+
+    fun removeRed() {
```

Comment:
Yes, that's why I immediately call `toOtherType(::AudioRtpPacket)` after using it. I didn't see a way around this.

Reply:
I was thinking we did something different for `RTX`--and we do--but I'm not sure it would work here (there is no concrete `RtxPacket` type, just static helper methods we use).  We could have `removeRed` create the `AudioRtpPacket` and return it?

Reply:
This would work for `removeRed`, but not for `removeRedAndGetRedundancyPackets` since it already returns a value.

Reply:
We handle RTX in the receive pipeline alone, so we can get away with recognizing it based on the payload type. We need to recognize RED packets in the send pipelines of endpoints without RED support.

Reply:
> This would work for `removeRed`, but not for `removeRedAndGetRedundancyPackets` since it already returns a value.

Yeah we could returning a pair there and destructuring would work pretty well I think:

```val (origPacket, redundantPackets) = removeRedAndGetRedundancyPackets(...)```

But I don't think I feel strongly...we can see how it feels and tweak later if we want.